x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java

```diff
@@ -540,6 +544,25 @@ public void testGetApiKeysOwnedByCurrentAuthenticatedUser() throws InterruptedEx
             response, userWithManageApiKeyRoleApiKeys.stream().map(o -> o.getId()).collect(Collectors.toSet()), null);
     }
 
+    public void testGetApiKeysOwnedByRunAsUser() throws ExecutionException, InterruptedException {
+        int noOfSuperuserApiKeys = randomIntBetween(3, 5);
+        int noOfApiKeysForUserWithManageApiKeyRole = randomIntBetween(3, 5);
+        createApiKeys(noOfSuperuserApiKeys, null);
+        final String userWithManageOwnApiKeyRole = "user_with_manage_own_api_key_role";
+        final String userWithRunAsRole = "user_with_run_as_role";
+        List<CreateApiKeyResponse> userWithManageOwnApiKeyRoleApiKeys = createApiKeys(userWithManageOwnApiKeyRole,
+            userWithRunAsRole, noOfApiKeysForUserWithManageApiKeyRole, null, "monitor");
+        final Client client = client().filterWithHeader(Map.of("Authorization", UsernamePasswordToken
+            .basicAuthHeaderValue(userWithRunAsRole, SecuritySettingsSourceField.TEST_PASSWORD_SECURE_STRING),
+            "es-security-runas-user", userWithManageOwnApiKeyRole));
+
+        PlainActionFuture<GetApiKeyResponse> listener = new PlainActionFuture<>();
+        client.execute(GetApiKeyAction.INSTANCE, GetApiKeyRequest.forOwnedApiKeys(), listener);
```

Comment:
add a test when the `owned` flag is not set, but the `username` and `realm` are.

Reply:
I was meant do test without the "owner" flag since the updated logic won't get executed with its presense. Somehow I ended up forgetting it. Will update. Thanks

Reply:
Add tests for [get](https://github.com/elastic/elasticsearch/pull/52178/files#diff-f1e82384e2498269e87f0e84280c8dbcR563) and [invalidate](https://github.com/elastic/elasticsearch/pull/52178/files#diff-f1e82384e2498269e87f0e84280c8dbcR670) API key without "owner=true"